### PR TITLE
FINISHED dtos expanding 🎉🥳🎁🎈✨

### PR DESCRIPTION
**Pull request by Evgeniy-Sinicin**

### CODE REVIEW

File: InternshipBE/BL/Services/EvaluationService.cs [hunk 1]
```diff
@@ -20,26 +19,35 @@ public EvaluationService(IUnitOfWork unitOfWork, IMapper mapper)
             _mapper = mapper;
         }
 
-        public async Task<CreateEvaluationDTO> CreateEvaluationAsync(CreateEvaluationDTO createEvaluationDto)
+        public async Task<EvaluationDTO> CreateEvaluationAsync(EvaluationDTO createEvaluationDto)
         {
-            var evaluation = await _unitOfWork.Evaluations.CreateAsync(_mapper.Map<Evaluation>(createEvaluationDto));
+            createEvaluationDto.Id = 0;
```
Comment (romanstat) [inline InternshipBE/BL/Services/EvaluationService.cs]: remove

File: InternshipBE/BL/Services/FeedbackService.cs [hunk 2]
```diff
@@ -19,32 +19,41 @@ public FeedbackService(IUnitOfWork unitOfWork, IMapper mapper)
             _mapper = mapper;
         }
 
-        public async Task<FullGetFeedbackDTO> CreateFeedbackAsync(CreateFeedbackDTO newFeedback)
+        public async Task<FeedbackDTO> CreateFeedbackAsync(FeedbackDTO newFeedback)
         {
-            var feedback = await _unitOfWork.Feedbacks.CreateAsync(_mapper.Map<Feedback>(newFeedback));
+            newFeedback.Id = 0;
```
Comment (romanstat) [inline InternshipBE/BL/Services/FeedbackService.cs]: remove

File: InternshipBE/BL/Services/FeedbackService.cs [hunk 3]
```diff
@@ -19,32 +19,41 @@ public FeedbackService(IUnitOfWork unitOfWork, IMapper mapper)
...
 
-            return _mapper.Map<FullGetFeedbackDTO>(feedback);
+            var feedback = _mapper.Map<Feedback>(newFeedback);
+
+            feedback = await _unitOfWork.Feedbacks.CreateAsync(feedback);
+            newFeedback = _mapper.Map<FeedbackDTO>(feedback);
+
+            return newFeedback;
```
Comment (romanstat) [inline InternshipBE/BL/Services/FeedbackService.cs]: return  _mapper.Map<FeedbackDTO>(feedback);

File: InternshipBE/BL/Services/FeedbackService.cs [hunk 4]
```diff
@@ -19,32 +19,41 @@ public FeedbackService(IUnitOfWork unitOfWork, IMapper mapper)
...
         }
 
-        public async Task<GetFeedbackDTO> GetFeedbackByIdAsync(int id)
+        public async Task<FeedbackDTO> GetFeedbackByIdAsync(int id)
         {
             var feedback = await _unitOfWork.Feedbacks.GetByIdAsync(id);
+            var feedbackDto = _mapper.Map<FeedbackDTO>(feedback);
 
-            return _mapper.Map<GetFeedbackDTO>(feedback);
+            return feedbackDto;
```
Comment (romanstat) [inline InternshipBE/BL/Services/FeedbackService.cs]: same

File: InternshipBE/BL/Services/FeedbackService.cs [hunk 5]
```diff
@@ -19,32 +19,41 @@ public FeedbackService(IUnitOfWork unitOfWork, IMapper mapper)
...
+        public async Task<FeedbackDTO> UpdateFeedbackAsync(FeedbackDTO updatedFeedback)
         {
-            var feedback = await _unitOfWork.Feedbacks.UpdateAsync(_mapper.Map<Feedback>(updatedFeedback));
+            var feedback = _mapper.Map<Feedback>(updatedFeedback);
+            
+            feedback = await _unitOfWork.Feedbacks.UpdateAsync(feedback);
+            updatedFeedback = _mapper.Map<FeedbackDTO>(feedback);
 
-            return _mapper.Map<UpdateFeedbackDTO>(feedback);
+            return updatedFeedback;
```
Comment (romanstat) [inline InternshipBE/BL/Services/FeedbackService.cs]: same

File: InternshipBE/BL/Services/InternshipService.cs [hunk 6]
```diff
@@ -19,36 +19,38 @@ public InternshipService(IUnitOfWork unitOfWork, IMapper mapper)
             _mapper = mapper;
         }
 
-        public async Task<UpdateInternshipDTO> CreateInternshipAsync(CreateInternshipDTO newInternship)
+        public async Task<InternshipDTO> CreateInternshipAsync(InternshipDTO newInternship)
         {
+            newInternship.Id = 0;
```
Comment (romanstat) [inline InternshipBE/BL/Services/InternshipService.cs]: remove

File: InternshipBE/BL/Services/InternshipService.cs [hunk 7]
```diff
@@ -19,36 +19,38 @@ public InternshipService(IUnitOfWork unitOfWork, IMapper mapper)
...
         }
 
-        public async Task<List<GetInternshipDTO>> GetInternshipsAsync(int pageSize, int pageNumber)
+        public async Task<List<InternshipDTO>> GetInternshipsAsync(int pageSize, int pageNumber)
         {
             var internships = await _unitOfWork.Internships.GetInternshipsAsync(pageSize, pageNumber);
-            var internshipDtos = _mapper.Map<List<GetInternshipDTO>>(internships);
+            var internshipDtos = _mapper.Map<List<InternshipDTO>>(internships);
 
             return internshipDtos;
```
Comment (romanstat) [inline InternshipBE/BL/Services/InternshipService.cs]: same

File: InternshipBE/BL/Services/InternshipService.cs [hunk 8]
```diff
@@ -19,36 +19,38 @@ public InternshipService(IUnitOfWork unitOfWork, IMapper mapper)
...
         }
 
-        public async Task<GetInternshipDTO> GetInternshipByIdAsync(int id)
+        public async Task<InternshipDTO> GetInternshipByIdAsync(int id)
         {
             var internship = await _unitOfWork.Internships.GetByIdAsync(id);
-            var internshipDto = _mapper.Map<GetInternshipDTO>(internship);
+            var internshipDto = _mapper.Map<InternshipDTO>(internship);
 
             return internshipDto;
```
Comment (romanstat) [inline InternshipBE/BL/Services/InternshipService.cs]: same

File: InternshipBE/BL/Services/InternshipStackService.cs [hunk 9]
```diff
@@ -21,11 +21,11 @@ public InternshipStackService(IUnitOfWork unitOfWork, IMapper mapper)
 
         public async Task<InternshipStackDTO> CreateInternshipStackAsync(InternshipStackDTO internshipStackDto)
         {
+            internshipStackDto.Id = 0;
```
Comment (romanstat) [inline InternshipBE/BL/Services/InternshipStackService.cs]: remove

File: InternshipBE/BL/Services/LocationService.cs [hunk 10]
```diff
@@ -18,15 +18,17 @@ public LocationService(IMapper mapper, IUnitOfWork unitOfWork)
...
         {
             var locations = await _unitOfWork.Locations.GetLocationsAsync();
 
-            return _mapper.Map<List<GetCountryDTO>>(locations);
+            return _mapper.Map<List<CountryDTO>>(locations);
         }
 
         public async Task<CountryDTO> CreateLocationAsync(CountryDTO locationName)
         {
+            locationName.Id = 0;
```
Comment (romanstat) [inline InternshipBE/BL/Services/LocationService.cs]: remove

File: InternshipBE/WebApi/Controllers/SkillController.cs [hunk 11]
```diff
@@ -25,9 +24,9 @@ public async Task<IActionResult> CreateSkill([FromBody] SkillDTO skillDto)
         }
 
         [HttpPut("updateSkill")]
-        public async Task<IActionResult> UpdateSkill([FromBody] FullSkillDTO fullSkillDto)
+        public async Task<IActionResult> UpdateSkill([FromBody] SkillDTO SkillDTO)
```
Comment (romanstat) [inline InternshipBE/WebApi/Controllers/SkillController.cs]: camel case

Comment (romanstat) [reply InternshipBE/WebApi/Controllers/SkillController.cs]: code convention, skill or skillDto, updatedSkill, use same everywhere

File: InternshipBE/DAL/Repositories/InternshipStackRepository.cs [hunk 12]
```diff
@@ -14,6 +14,25 @@ public InternshipStackRepository(ApplicationDbContext context) : base(context)
...
+                }
+                else
+                {
+                    newIntershipStacks.Add(await CreateAsync(internshipStack));
+                }
+            }
+
+            return newIntershipStacks;
+        }
+
```
Comment (romanstat) [inline InternshipBE/DAL/Repositories/InternshipStackRepository.cs:35]: need to discuss

**Review by romanstat — APPROVED** [review]

**Review by Xodoc — APPROVED** [review]

**Review by ihhyy — APPROVED** [review]

**Pull request merged**